src/datasets/demo_only.py

```diff
@@ -20,6 +23,73 @@ def create_demo_data(env: BaseEnv, train_tasks: List[Task],
+    dataset_fname = os.path.join(
+        CFG.data_dir,
+        dataset_fname_template.replace(regex, str(CFG.num_train_tasks)))
+    os.makedirs(CFG.data_dir, exist_ok=True)
+    if CFG.load_data:
+        # Handle loading data from disk in three separate cases.
+        if os.path.exists(dataset_fname):
+            # Case 1: we already have a file with the exact name that we need
+            # (i.e., the correct amount of data).
+            with open(dataset_fname, "rb") as f:
+                dataset = pkl.load(f)
+            logging.info(f"LOADED DATASET OF {len(dataset.trajectories)} "
+                         "DEMONSTRATIONS")
+            return dataset
+        fnames_with_less_data = {}  # used later, in Case 3
+        for fname in os.listdir(CFG.data_dir):
+            regex_match = re.match(dataset_fname_template, fname)
+            if not regex_match:
+                continue
+            num_train_tasks = int(regex_match.groups()[0])
+            assert num_train_tasks != CFG.num_train_tasks  # would be Case 1
+            # Case 2: we already have a file with MORE data than we need. Load
+            # and truncate this data.
+            if num_train_tasks > CFG.num_train_tasks:
+                with open(os.path.join(CFG.data_dir, fname), "rb") as f:
+                    dataset = pkl.load(f)
+                logging.info("LOADED AND TRUNCATED DATASET OF "
+                             f"{len(dataset.trajectories)} DEMONSTRATIONS")
+                assert not dataset.has_annotations
+                return Dataset(dataset.trajectories[:CFG.num_train_tasks])
+            # Save the names of all datasets that have less data than
+            # we need, to be used in Case 3.
+            fnames_with_less_data[num_train_tasks] = fname
+        if not fnames_with_less_data:
+            # Give up: we did not find any data file we can load from.
+            raise ValueError(f"Cannot load data: {dataset_fname}")
+        # Case 3: we already have a file with LESS data than we need. Load
+        # this data and generate some more.
+        start_idx = max(fnames_with_less_data)
+        fname = fnames_with_less_data[start_idx]
+        with open(os.path.join(CFG.data_dir, fname), "rb") as f:
+            dataset = pkl.load(f)
+        loaded_trajectories = dataset.trajectories
+        logging.info(f"LOADED DATASET OF {len(loaded_trajectories)} "
+                     "DEMONSTRATIONS")
+    else:
+        # We are not using `--load_data`, so we must generate all needed data.
+        start_idx = 0
+        loaded_trajectories = []
+    # Now, generate any necessary amount of data. We can only get here if either
+    # we're in Case 3 of `--load_data` being used, or `--load_data` is not used.
+    generated_trajectories = _create_trajs_from_start_idx(
+        env, train_tasks, known_options, start_idx)
+    logging.info(f"CREATED {len(generated_trajectories)} DEMONSTRATIONS")
```

Comment:
for extra clarity, add one more log info reporting the final dataset size?

Reply:
i chose not to bc this will already be printed at the start of NSRT learning, for instance the last line here:
```
LOADED DATASET OF 10 DEMONSTRATIONS
Planning succeeded! Found plan of length 1 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 3 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 1 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 1 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 3 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 3 after 2 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 3 after 2 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 3 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 1 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 1 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 3 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 1 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 2 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 3 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 2 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 1 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 4 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 2 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 3 after 2 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 3 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 2 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 3 after 2 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 3 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 3 after 1 skeletons, discovering 0 failures
Planning succeeded! Found plan of length 2 after 1 skeletons, discovering 0 failures
CREATED 25 DEMONSTRATIONS

Learning NSRTs on 35 trajectories...
```

Reply:
got it, makes sense